src/main/java/org/apache/commons/compress/compressors/z/ZCompressorInputStream.java

```diff
@@ -72,6 +72,12 @@ public ZCompressorInputStream(final InputStream inputStream, final int memoryLim
         if (blockMode) {
             setClearCode(DEFAULT_CODE_SIZE);
         }
+        // maxCodeSize cannot be zero, otherwise initializeTables() will throw an IllegalArgumentException
+        // maxCodeSize shifted cannot be less than 256, otherwise the loop in initializeTables() will throw an ArrayIndexOutOfBoundsException
+        // maxCodeSize cannot be smaller than getCodeSize(), otherwise addEntry() will throw an ArrayIndexOutOfBoundsException
+        if (maxCodeSize <= 0 || (1 << maxCodeSize) < 256 || getCodeSize() > maxCodeSize) {
+            throw new IOException("Invalid maxCodeSize value: " + maxCodeSize);
```

Comment:
Hello @yakovsh 
Thank you for your update. I still don't understand the point of this change. Without this check, the code throws `java.lang.IllegalArgumentException: maxCodeSize is 0, must be bigger than 0` from `LZWInputStream.initializeTables(int, int)` . So why do we need this check instead of the one in `initializeTables()`?

The test seems fine.